src/sync/impl/sync_client.hpp

```diff
@@ -24,7 +24,19 @@
 #include <thread>
 
 namespace realm {
-namespace _impl {
+
+// Interface for bindings interested in the lifecycle of the Sync Client thread.
+class ClientThreadListener {
```

Comment:
From looking at the Java PR that builds on top of this, it doesn't look like there's anything specific to the sync client in this callback. It might be preferable to give this a name that's more general. If I understand correctly this pair of calls needs to be made on any thread created by the object store that may call into Java?